style(tables): FRMW-321 Update Tables to Color 2.0

ssalinas24 opened the pull request:
JIRA: https://jira.rax.io/browse/FRMW-321

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM

I received Feedback from Eric in the comments below, I made my changes in my second commit, and then updated this initial comment with the latest screenshots.

## Loading Overlay
### Before
![00_before-loading-overlay](https://cloud.githubusercontent.com/assets/14296817/11542406/cf89f322-98fc-11e5-9cf7-ade0e9d19662.png)

### After
![00_after-loading-overlay](https://cloud.githubusercontent.com/assets/14296817/11699895/559d52da-9e8c-11e5-9c1c-79f63f911746.png)


## Tables Basics
### Before
![01_before-tabels-basics](https://cloud.githubusercontent.com/assets/14296817/11542423/dc945fbc-98fc-11e5-8f70-8270c475eb07.png)

### After
![01_after-tabels-basics](https://cloud.githubusercontent.com/assets/14296817/11862217/0bb2410e-a44e-11e5-8269-ee5799f5415a.png)

## rxPaginate
### Before
![02_before-rxpaginate](https://cloud.githubusercontent.com/assets/14296817/11542437/ec62c2d0-98fc-11e5-93d8-ff55737563eb.png)

### After
<img width="1046" alt="02_after-rxpaginate" src="https://cloud.githubusercontent.com/assets/14296817/11862226/2b878818-a44e-11e5-99a6-05cc93264770.png">

## Delete X Action
### Before
![03_before-delete-x-action](https://cloud.githubusercontent.com/assets/14296817/11542461/0253e2a4-98fd-11e5-9ff5-cdd20a96f08d.png)

### After
<img width="1048" alt="03_after-delete-x-action" src="https://cloud.githubusercontent.com/assets/14296817/11862227/378d47f6-a44e-11e5-8a3a-7eba881c8ebb.png">


## Single Action
### Before
![04_before-single-action](https://cloud.githubusercontent.com/assets/14296817/11542475/1749ddbc-98fd-11e5-947c-ce00fbc247ad.png)

### After
<img width="1047" alt="04_after-single-action" src="https://cloud.githubusercontent.com/assets/14296817/11862230/41033f5c-a44e-11e5-861b-ef497337fcb7.png">

## Filters
### Before
![05_before-filters](https://cloud.githubusercontent.com/assets/14296817/11542481/21343ed0-98fd-11e5-83de-8eeee813f671.png)

### After
<img width="1048" alt="05_after-filters" src="https://cloud.githubusercontent.com/assets/14296817/11862234/48ee8672-a44e-11e5-8326-d308596a6fa2.png">

## rxCollapse
### Before
![06_before-rxcollapse](https://cloud.githubusercontent.com/assets/14296817/11542485/2de23682-98fd-11e5-8ac0-878244c1209e.png)

### After
<img width="1045" alt="06_after-rxcollapse" src="https://cloud.githubusercontent.com/assets/14296817/11862236/51c9eb42-a44e-11e5-8053-e97cab60ff38.png">

## Expanded
### Before
![07_before-expanded](https://cloud.githubusercontent.com/assets/14296817/11542495/3a925330-98fd-11e5-8245-973ddf83cbb3.png)

### After
<img width="1044" alt="07_after-expanded" src="https://cloud.githubusercontent.com/assets/14296817/11862239/5c137f14-a44e-11e5-855e-0f235f0e9de6.png">

## Metadata
### Before
![08_before-metadata](https://cloud.githubusercontent.com/assets/14296817/11542500/4540b5c4-98fd-11e5-932d-d57d3ed45860.png)

### After
<img width="1048" alt="08_after-metadata" src="https://cloud.githubusercontent.com/assets/14296817/11862242/656225f2-a44e-11e5-8bbf-301aafc54402.png">

## rxPaginate API Based
### Before
![10_before-rxpaginate-api-based_tota11y](https://cloud.githubusercontent.com/assets/14296817/11542512/55eeafc0-98fd-11e5-9fa8-3366193682b4.png)

### After
<img width="1046" alt="10_after-rxpaginate-api-based" src="https://cloud.githubusercontent.com/assets/14296817/11862243/6dd2ac8e-a44e-11e5-825e-6ca782d464a2.png">

## rxBulkSelect
### Before
![11_before-rxbulkselect](https://cloud.githubusercontent.com/assets/14296817/11542521/64aa407e-98fd-11e5-8b6a-0b633fe6b4da.png)

### After
<img width="1046" alt="11_after-rxbulkselect" src="https://cloud.githubusercontent.com/assets/14296817/11862248/75c9d0b6-a44e-11e5-8274-dbe326efe210.png">

## rxOptionTable
### Before
![12_before-rxoptiontable](https://cloud.githubusercontent.com/assets/14296817/11542533/7378ed26-98fd-11e5-8d8a-3def85337676.png)

### After
<img width="625" alt="12_after-rxoptiontable" src="https://cloud.githubusercontent.com/assets/14296817/11862255/82cd8474-a44e-11e5-958f-68be1a383dd3.png">

## rxFloatingHeader
### Before
![14_before-rxfloatingheader](https://cloud.githubusercontent.com/assets/14296817/11542557/8c61a72e-98fd-11e5-9971-efadb968c696.png)

### After
<img width="1043" alt="14_after-rxfloatingheader-01" src="https://cloud.githubusercontent.com/assets/14296817/11862266/960234cc-a44e-11e5-8ab5-768fba29fb42.png">

## rxSortableColumn
This was updated after James' PR, then updated the colors here.

### Before
![15_before-rxsortablecolumn](https://cloud.githubusercontent.com/assets/14296817/11909218/55ecba36-a5a9-11e5-9de5-e8b7ff5163b1.png)

### After
![15_after-rxsortablecolumn](https://cloud.githubusercontent.com/assets/14296817/11909221/5ae5f91c-a5a9-11e5-93ea-fb33bc1b438a.png)

ssalinas24 commented:
@ElementE I copied the problem images from above and put them in this comment with my notes and questions. Below are the items that I will need your feedback on.

## Loading Overlay
This shows loading overlay with `@gray-700` for the spinner and text color falling below the contrast ratio. It doesn't meet the contrast ratio. Do you want to darken the foreground text/spinner or lighten the background?

![00_after-loading-overlay](https://cloud.githubusercontent.com/assets/14296817/11517195/e2983fdc-984d-11e5-9c50-2b2eccec0821.png)

### Before
![00_before-loading-overlay](https://cloud.githubusercontent.com/assets/14296817/11537940/268a4a54-98e5-11e5-8b4b-d7850d702b0f.png)

## rxPaginate
The "Next" is shown here in the hover state with a `@gray-50` as a background. The contrast ratio is included in another screen down below. I left it off of this one so you could see the area clearly. The same `@gray-50` is used in the alternating table row backgrounds. Do we just make the pagination text to be the same as the table rows?

![02_after-rxpaginate](https://cloud.githubusercontent.com/assets/14296817/11517196/e299e6c0-984d-11e5-85ed-30e5dce487bb.png)

As mentioned above on the paginate image. This one just shows the actual contrast ratios.
![09_after-rxpaginate-tota11y](https://cloud.githubusercontent.com/assets/14296817/11517201/e2a4e8e0-984d-11e5-821d-4617fb4cbcff.png)

### Before
![02_before-rxpaginate_tota11y](https://cloud.githubusercontent.com/assets/14296817/11537970/42f51e12-98e5-11e5-8744-2257695fee06.png)

![02_before-rxpaginate](https://cloud.githubusercontent.com/assets/14296817/11537969/42f2e7f0-98e5-11e5-936a-b2f51d71b936.png)

## Single Action
This shows a contrast ratio problem with blue and disabled links on the table rows. The blue link is already using `@blue-700`. Eric what are your thoughts on solving this one?

![04_after-single-action](https://cloud.githubusercontent.com/assets/14296817/11517198/e29dfb5c-984d-11e5-9178-660f7e3ce5e0.png)

### Before
![04_before-single-action](https://cloud.githubusercontent.com/assets/14296817/11538009/71f013a2-98e5-11e5-9713-6450818ace42.png)

## rxCollapse
No real contrast issues on this one, but i wanted to confirm the top gray. Do we still continue to keep the lighter gray and white double chevron arrow as shown here?

![06_after-rxcollapse](https://cloud.githubusercontent.com/assets/14296817/11517199/e2a2cf6a-984d-11e5-82ff-f125e000e474.png)

### Before
![06_before-rxcollapse](https://cloud.githubusercontent.com/assets/14296817/11538031/927e476a-98e5-11e5-9ad9-bcc1b1cbb14c.png)

## Expanded
The sub table had issues with contrast ratios on the table header gray background with white text. So I decided to deliver this option of using the same colors from the parent table. It uses the same dark gray, white text, and border colors. Eric would you agree with this, or do you have something else in mind?

![07_after-expanded](https://cloud.githubusercontent.com/assets/14296817/11517200/e2a4ad58-984d-11e5-9e9a-57694eec970d.png)

### Before
![07_before-expanded](https://cloud.githubusercontent.com/assets/14296817/11538044/a685f7e4-98e5-11e5-9036-7d998db5a047.png)

## Metadata
Same note as mentioned on the previous "Expanded" image above.

![08_after-metadata](https://cloud.githubusercontent.com/assets/14296817/11517202/e2a5c508-984d-11e5-8592-7612855a4fc7.png)

### Before
![08_before-metadata](https://cloud.githubusercontent.com/assets/14296817/11538054/b264c518-98e5-11e5-82d2-26b114a59c40.png)

## rxBulkSelect
This shows contrast ratio issues with the blue link text on green background. The blue link is already using the darkest `@blue-700`. The green is also using the lightest `@green-100`. Eric what should we do? Darken the link color? Lighten the green background? or any other ideas?

![11_after-rxbulkselect](https://cloud.githubusercontent.com/assets/14296817/11517204/e2ab5252-984d-11e5-8bab-738c4c1ee4d2.png)

### Before
![11_before-rxbulkselect](https://cloud.githubusercontent.com/assets/14296817/11538071/c55b7f22-98e5-11e5-9388-bf75731a68c5.png)


## rxOptionTable
No contrast ratio issues with the green, but it is a little too similar to the dark grays. It is using the darkest  `@green-700`, but is the only one that passes the color contrast ratio. We will have to keep for now, but may want to keep this in mind for later. The gray disabled rows are using the colors that we have established, so this is how it would look. Any problems with keeping this?

For the green selected rows on the bottom image, notice that the borders are a little brighter then before. I used `@green-500`. Is it okay to proceed like this or preferred to dim them?

![12_after-rxoptiontable](https://cloud.githubusercontent.com/assets/14296817/11517205/e2ade85a-984d-11e5-9031-4f6b021ea3e8.png)

### Before
![12_before-rxoptiontable](https://cloud.githubusercontent.com/assets/14296817/11538084/d9e80366-98e5-11e5-890b-aa776f925189.png)


## rxSortableColumn
The arrow is a little dark. Is this correct to proceed? I also noticed that in some cases the arrow is white? Are we supposed to stick with a specific color?

![13_after-rxsortablecolumn](https://cloud.githubusercontent.com/assets/14296817/11517207/e2b1e180-984d-11e5-917c-e08f587dc2ba.png)

### Before
![13_before-rxsortablecolumn](https://cloud.githubusercontent.com/assets/14296817/11538098/ea65afa4-98e5-11e5-87bd-1b8442a235e4.png)

ElementE commented:
Loading Overlay 
-lets lighten the BG to increase the contrast


Tables Basics
- Lets shift table header BG to #E7E7E7, header titles to #616161 
(contrast ratio is 5.00832)


Pagination
- i say we keep the text color as is, if anything we can introduce a darker/colored hover for those lower actions (i.e., “next”)


Single Action
- blue looks fine, i think the shift from disabled gives enough contrast overall. 

rx-Collapse
- LGTM, yes lets keep the top grey for the chevron

expanded/metadata 
- i think we should look at changing the child row -expanded wrapper color, ideally something lighter than the main table header. I will revisit, but also open to any suggestions.


rxBulkSelect
- So i think with areas like highlighting rows maybe we should explore an opacity value given to the selection BG (in this case, the green). Maybe lowering to 60-70%? 


rxOptionTable
- (Radios) Lets leave this for now
- (checkbox) I think we should give them the lowered opacity as stated above in BulkSelect


rxSortableColumn 
- this should now be solved by the table-header color shift (given above in Table Basics)

ssalinas24 commented on src/components/rxPaginate/rxPaginate.less:
```diff
@@ -114,7 +113,8 @@
   .pagination > .page-links > li > a:focus,
   .pagination > .page-links > li > span:focus {
-    background: #eeeeee;
+    border-top: 2px solid @gray-300;
```
@glynnis , @ElementE  and I were discussing not having a background on the pagination. I tried the darker gray background on the pagination number when on hover. However, it doesn't look clean and elegant. So I decided to remove the background and just use a dark gray border on top upon the hover state. Do you guys think this will work?

It happens on hover for the "Next" and "Last" links as well.

glynnis replied:
I think no background is fine :+1: If you think it needs a little somethin' extra, you could go with `@blue-100` background on hover as we've done on some other "you're about to select this"-type treatments.

CITguy commented:
FYI: #1435 also makes some modifications to rxSortableColumn, so depending on which one of the PRs is merged first, the other will need to be updated accordingly.

ElementE commented:
![table-specs-updated](https://cloud.githubusercontent.com/assets/7989545/11815656/4b8e52f0-a312-11e5-9f29-e3de603da3fd.png)

ElementE commented:
^ Changes with Red * are the latest. Summary: Darkening, table headers, lightening bg and zebra stripes.

ElementE commented:
![screen shot 2015-12-15 at 10 05 44 am](https://cloud.githubusercontent.com/assets/7989545/11815909/95d33f1e-a313-11e5-87aa-ec2acacfc805.png)
Also - Roboto-Medium for Th font

glynnis commented on src/styles/vars.less:
```diff
@@ -76,18 +76,17 @@
-@subtableBorder: #adadad;
-@tableHeaderText: @white;
-@tableCellText: #555555;
+@tableBgAlt: @gray-50;
```
Per @ElementE's mock-ups, change to `@gray-25`

glynnis commented on src/styles/vars.less:
```diff
@@ -76,18 +76,17 @@
+@tableBorder: @gray-300;
+@tableHeaderBorder: @gray-400;
+@subtableBorder: @gray-500;
+@tableHeaderText: @gray-800;
```
Per @ElementE's mock-ups, this should change to `@gray-900`.

glynnis commented on src/styles/vars.less:
```diff
@@ -76,18 +76,17 @@
+@tableHeaderBorder: @gray-400;
+@subtableBorder: @gray-500;
+@tableHeaderText: @gray-800;
+@tableCellText: @gray-900;
```
Per @ElementE's mock-ups, this should change to @gray-800.

glynnis commented:
@ssalinas24 I know it's not color-related, but while we're in here fixing styles, could you add some some padding to the "After" Metadata example of the expanded table row (without affecting the normal table cell rows in the expanded view)?

Similarly, the "Empty Option Table" content row lacks padding. I know I've seen examples of the "no data" treatment where the padding is correct, so I'm wondering if there's just a class missing there.

ssalinas24 commented:
@glynnis and @ElementE I have updated this PR to include changes for the feedback. I have updated the initial comment screenshots to reflect the most recent changes.

glynnis commented:
LGTM

CITguy commented on src/components/rxApp/common.less:
```diff
@@ -1,7 +1,7 @@
  * Common
  */
-@import url("https://fonts.googleapis.com/css?family=Roboto:400,100,100italic,300,300italic,400italic,700,700italic");
+@import url("https://fonts.googleapis.com/css?family=Roboto:400,100,100italic,300,300italic,400italic,700,700italic,500");
```
If we're including `500`, we might as well bring in `500italic`, too. Please update.

CITguy commented on src/components/rxOptionTable/rxOptionTable.less:
```diff
@@ -7,7 +7,7 @@
   min-width: 39em;
 
-  td:not(.empty-data) {
+  td:not(.empty-message) {
```
:ok:

100stacks commented:
Great job @ssalinas24 @ElementE! :+1: 

LGTM

ssalinas24 commented on src/components/rxApp/common.less:
```diff
@@ -1,7 +1,7 @@
  * Common
  */
-@import url("https://fonts.googleapis.com/css?family=Roboto:400,100,100italic,300,300italic,400italic,700,700italic");
+@import url("https://fonts.googleapis.com/css?family=Roboto:400,100,100italic,300,300italic,400italic,500,500italic,700,700italic");
```
@CITguy Good idea. I updated the import to include the 500italic as well.

bry4n commented:
LGTM. can you rebase?

CITguy commented:
@ssalinas24, please rebase and address colors for the new rxSortableColumn icons introduced in #1435.

CITguy commented:
LGTM

CITguy commented:
Looks awesome!
